sklearn/datasets/_base.py

```diff
@@ -280,6 +285,67 @@ def load_data(module_path, data_file_name):
     return data, target, target_names
 
 
+def load_gzip_compressed_csv_data(data_module, data_file_name, **kwargs):
```

Comment:
I don't think we can ask `np.loadtxt` to decompress if we are passing in file object (as opposed to a file name ending with '.gz'). So for the few `.csv.gz` files, we first manually decompress the file object and the use `np.loadtxt` to load data as before.